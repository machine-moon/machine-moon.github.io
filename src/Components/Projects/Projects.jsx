import PropTypes from "prop-types";
import "./Projects.css";

import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";
import project1_img from "../../assets/project1.jpg";
import project2_img from "../../assets/project2.png";
import project3_img from "../../assets/project3.png";
import project4_img from "../../assets/project4.jpg";
import project5_img from "../../assets/project5.jpg";
import project6_img from "../../assets/project6.png";

const ProjectCard = ({ work }) => (
  <div className="project-card">
    <img src={work.w_img} alt={work.w_name} />
    <div className="project-overlay">
      <h2>{work.w_name}</h2>
      <p>{work.w_description}</p>
      <a href={work.w_link} target="_blank" rel="noopener noreferrer">
        <button className="project-button">View Project</button>
      </a>
    </div>
  </div>
);

ProjectCard.propTypes = {
  work: PropTypes.shape({
    w_img: PropTypes.string.isRequired,
    w_name: PropTypes.string.isRequired,
    w_description: PropTypes.string.isRequired,
    w_link: PropTypes.string.isRequired,
  }).isRequired,
};

const Projects = () => {
  return (
    <div id="projects" className="mywork">
      <div className="mywork-title">
        <h1>My Projects</h1>
      </div>
      <div className="mywork-container">
        {ProjectData.map((work, index) => (
          <ProjectCard key={index} work={work} />
        ))}
      </div>
      <div className="mywork-showmore">
        <a
          href="https://github.com/machine-moon?tab=repositories"
          target="_blank"
          rel="noopener noreferrer"
        >
          <p>More Projects</p>
        </a>
        <div className="arrow-outward-icon">
          <ArrowForwardIosIcon />
        </div>
      </div>
    </div>
  );
};

const ProjectData = [
  {
    w_no: 1,
    w_name: "TDocker",
    w_img: project1_img,
    w_description:
      "A streamlined suite of Docker Compose setups for easy application deployment on an NGINX Network",
    w_link: "https://github.com/machine-moon/tdocker",
  },
  {
    w_no: 2,
    w_name: "TSuite",
    w_img: project2_img,
    w_description:
      "A collection of workflow tools and scripts designed to automate tasks and boost productivity",
    w_link: "https://github.com/machine-moon/tsuite",
  },
  {
    w_no: 3,
    w_name: "Markov Decision Process Simulator",
    w_img: project3_img,
    w_description:
      "A very fast simulator for MDPs using JAX, featuring JIT compilation for rapid prototyping and reinforcement learning with a ready-to-use DQN algorithm",
    w_link: "https://github.com/machine-moon/mdp-jax",
  },
  {
    w_no: 4,
    w_name: "Scrabble",
    w_img: project4_img,
    w_description:
      "A Java-based Scrabble game utilizing MVC architecture, offering interactive gameplay with a GUI and Bot AI",
    w_link: "https://github.com/machine-moon/project4",
  },
  {
    w_no: 5,
    w_name: "Embedded Microcontroller Projects",
    w_img: project5_img,
    w_description:
      "Showcasing expertise in embedded systems, focusing on interrupts, NVIC, Scheduling, GPIO, timers, and state machines. Using MSVP32 AND rasperry pi",
    w_link: "https://github.com/machine-moon/project5",
  },
  {
    w_no: 6,
    w_name: "Personal Portfolio Website",
    w_img: project6_img,
    w_description:
      "A personal portfolio website using React.js, html, css, and javascript, and deployed it using GitHub Pages",
    w_link: "https://github.com/machine-moon/project6",
  },
];

export default Projects;
