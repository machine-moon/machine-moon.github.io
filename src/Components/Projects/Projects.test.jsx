import { test, expect } from '@jest/globals';
import { render } from '@testing-library/react';
import Projects from './Projects';

test('renders Projects component correctly', () => {
  const { asFragment } = render(<Projects />);
  expect(asFragment()).toMatchSnapshot();
});
