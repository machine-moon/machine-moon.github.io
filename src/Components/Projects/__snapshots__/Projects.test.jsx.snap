// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Projects component correctly 1`] = `
<DocumentFragment>
  <div
    class="mywork"
    id="projects"
  >
    <div
      class="mywork-title"
    >
      <h1>
        My Projects
      </h1>
    </div>
    <div
      class="mywork-container"
    >
      <div
        class="project-card"
      >
        <img
          alt="TDocker"
          src="test-file-stub"
        />
        <div
          class="project-overlay"
        >
          <h2>
            TDocker
          </h2>
          <p>
            A streamlined suite of Docker Compose setups for easy application deployment on an NGINX Network
          </p>
          <a
            href="https://github.com/machine-moon/tdocker"
            rel="noopener noreferrer"
            target="_blank"
          >
            <button
              class="project-button"
            >
              View Project
            </button>
          </a>
        </div>
      </div>
      <div
        class="project-card"
      >
        <img
          alt="TSuite"
          src="test-file-stub"
        />
        <div
          class="project-overlay"
        >
          <h2>
            TSuite
          </h2>
          <p>
            A collection of workflow tools and scripts designed to automate tasks and boost productivity
          </p>
          <a
            href="https://github.com/machine-moon/tsuite"
            rel="noopener noreferrer"
            target="_blank"
          >
            <button
              class="project-button"
            >
              View Project
            </button>
          </a>
        </div>
      </div>
      <div
        class="project-card"
      >
        <img
          alt="Markov Decision Process Simulator"
          src="test-file-stub"
        />
        <div
          class="project-overlay"
        >
          <h2>
            Markov Decision Process Simulator
          </h2>
          <p>
            A very fast simulator for MDPs using JAX, featuring JIT compilation for rapid prototyping and reinforcement learning with a ready-to-use DQN algorithm
          </p>
          <a
            href="https://github.com/machine-moon/mdp-jax"
            rel="noopener noreferrer"
            target="_blank"
          >
            <button
              class="project-button"
            >
              View Project
            </button>
          </a>
        </div>
      </div>
      <div
        class="project-card"
      >
        <img
          alt="Scrabble"
          src="test-file-stub"
        />
        <div
          class="project-overlay"
        >
          <h2>
            Scrabble
          </h2>
          <p>
            A Java-based Scrabble game utilizing MVC architecture, offering interactive gameplay with a GUI and Bot AI
          </p>
          <a
            href="https://github.com/machine-moon/project4"
            rel="noopener noreferrer"
            target="_blank"
          >
            <button
              class="project-button"
            >
              View Project
            </button>
          </a>
        </div>
      </div>
      <div
        class="project-card"
      >
        <img
          alt="Embedded Microcontroller Projects"
          src="test-file-stub"
        />
        <div
          class="project-overlay"
        >
          <h2>
            Embedded Microcontroller Projects
          </h2>
          <p>
            Showcasing expertise in embedded systems, focusing on interrupts, NVIC, Scheduling, GPIO, timers, and state machines. Using MSVP32 AND rasperry pi
          </p>
          <a
            href="https://github.com/machine-moon/project5"
            rel="noopener noreferrer"
            target="_blank"
          >
            <button
              class="project-button"
            >
              View Project
            </button>
          </a>
        </div>
      </div>
      <div
        class="project-card"
      >
        <img
          alt="Personal Portfolio Website"
          src="test-file-stub"
        />
        <div
          class="project-overlay"
        >
          <h2>
            Personal Portfolio Website
          </h2>
          <p>
            A personal portfolio website using React.js, html, css, and javascript, and deployed it using GitHub Pages
          </p>
          <a
            href="https://github.com/machine-moon/project6"
            rel="noopener noreferrer"
            target="_blank"
          >
            <button
              class="project-button"
            >
              View Project
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="mywork-showmore"
    >
      <a
        href="https://github.com/machine-moon?tab=repositories"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p>
          More Projects
        </p>
      </a>
      <div
        class="arrow-outward-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
